### PR TITLE
implement Crispy HUD as an alternative to Boom HUD

### LINKED ISSUE #474: [feature suggestion] Crispy HUD
Hi, Fabian. Thanks for the port! Absolutely loving what you and co have done to give us such an amazing port.

The only thing I miss from Crispy Doom is its hud. IMO it's the best fullscreen hud a doom port can have.

I remember you responded someone who requested it that you hadn't included it because MBF already has Boom's fullscreen hud. Since the level stats widget has been implemented and almost everyone uses the regular status bar + this widget, I think Crispy hud could replace the outdated Boom hud if you don't want two fullscreen huds in your port.

It's just a suggestion. Will look forward what you think about it.

### PR DESCRIPTION
**Pull request by fabiangreffrath**
Fixes #474

### CODE REVIEW

File: Source/hu_stuff.c [hunk 1]
```diff
@@ -1764,7 +1779,7 @@ void HU_Ticker(void)
     if (hud_timests)
     {
       HU_widget_build_sttime();
-      if (scaledviewheight < SCREENHEIGHT)
+      if (scaledviewheight < SCREENHEIGHT || crispy_hud)
```
Comment (fabiangreffrath) [inline Source/hu_stuff.c:1782]: st_crispyhud

File: Source/hu_stuff.c [hunk 2]
```diff
@@ -1058,6 +1060,18 @@ void HU_Drawer(void)
...
+      {
+          ST_Drawer (false, true);
+          if (hud_timests)
+          {
+              HUlib_drawTextLine(&w_sttime, false);
+              HUlib_drawTextLine(&w_monsec, false);
+          }
+      }
+      else
+      {
```
Comment (fabiangreffrath) [inline Source/hu_stuff.c:1074]: line 1516

File: Source/m_menu.c [hunk 3]
```diff
@@ -3168,16 +3168,17 @@ setup_menu_t stat_settings2[] =
 
   {"EXTENDED HUD",S_SKIP|S_TITLE,m_null,ST_X,ST_Y+ 5*8 },
 
-  {"\"A SECRET IS REVEALED!\" MESSAGE",S_YESNO,m_null,ST_X,ST_Y+6*8, {"hud_secret_message"}},
-  {"SHOW TIME/STS ABOVE STATUS BAR",S_YESNO,m_null,ST_X,ST_Y+7*8, {"hud_timests"}},
+  {"PREFER CRISPY HUD OVER BOOM HUD",S_YESNO,m_null,ST_X,ST_Y+6*8, {"crispy_hud"}},
```
Comment (fabiangreffrath) [inline Source/m_menu.c:3171]: PREFER CRISPY OVER BOOM HUD

File: Source/st_stuff.c [hunk 4]
```diff
@@ -1164,38 +1179,54 @@ void ST_createWidgets(void)
...
                 ST_MAXAMMO3Y,
                 shortnum,
                 &plyr->maxammo[3],
                 &st_statusbaron,
                 ST_MAXAMMO3WIDTH);
 }
 
+static void ST_MoveHud (void)
+{
+    static int odelta = -1;
```
Comment (fabiangreffrath) [inline Source/st_stuff.c:1216]: = 0

File: Source/hu_stuff.c (hunk 1 above)
Comment (fabiangreffrath) [reply Source/hu_stuff.c:1782]: or (crispy_hud && hud_displayed)

**Comment by fabiangreffrath** [comment]
I don't like the floating face on front of the weapon sprite. In order to keep the number of choices reasonable, I'd suggest to go without the face first. People will complain about it anyway and then I can still decide whether I want to add this or not.

**Review by rfomin — APPROVED** [review]

**Pull request merged**